### DISCUSSION TITLE
서버 데이터 스토리지 관련 질문

### DISCUSSION THREAD

**Discussion by ted002** [naverz/zepeto-studio-kor#777] [5. Q&A]
import { DataStorage } from "ZEPETO.Multiplay.DataStorage";

// load current player's data storage
const playerStorage:DataStorage = client.loadDataStorage();

해당 playerStorage의 경우, 서버에 데이터를 저장한다고 되어있는데, 이는 서로 공유되지 않는 데이터인가요?

`if(userid == "A클라이언트userid")
{
await playerStorage.set("level", 1);
}
if(userid == "B클라이언트userid")
{
let a = await playerStorage.Get("level") as number;
consol.log(a)
}
`

라고 했을때, A클라이언트가 접속하고 B 클라이언트가 접속했을때 a는 undefined로 나오더군요

클라이언트별로 서버 데이터 스토리지 하나가 할당되는게 아니고, 클라이언트 하나에 서버 데이터 스토리지 하나가 할당되는 식인건가요?

**Comment by minjujuu** [comment]
`client.loadDataStorage();` 에서 client는 SandboxPlayer로 입장한 플레이어를 나타내서
입장한 플레이어의 DataStorage를 불러올텐데요

https://studio.zepeto.me/kr/guides/multiplay-game-datastorage
가이드를 보면 set한 적 없는 key를 get하게 되면 Null을 반환한다고 하는데, 
아마도 보여주신 코드에서는 A 클라이언트에 대한 level 정보만 set하고, 
B클라이언트에 대한 level 정보는 set 하는 코드 없이 get 하고 있어서 그런 것으로 보입니다.

말씀주신 것처럼 서버 데이터 스토리지는 클라이언트 하나에 서버 데이터 스토리지 하나가 할당되는 식으로 활용할 수 있을 것 같습니다!
